tests/bluetooth/bsim/nrf_auraconfig/tester/src/nrf_auraconfig_test.c

```diff
@@ -0,0 +1,1164 @@
+	}
+}
+
+void nac_test_loop(void)
+{
+	int err;
+	uint32_t sync_bitfield;
+
+	err = k_sem_take(&sem_broadcaster_found, K_FOREVER);
+	if (err != 0) {
+		LOG_DBG("sem_broadcaster_found timed out, resetting");
+	}
+
+	err = bt_le_scan_stop();
+	if (err != 0) {
+		LOG_DBG("bt_le_scan_stop failed with %d, resetting", err);
+	}
+
+	LOG_DBG("Attempting to PA sync to the broadcaster with id 0x%06X",
+		broadcaster_broadcast_id);
+
+	err = pa_sync_create();
+	if (err != 0) {
+		LOG_DBG("Could not create Broadcast PA sync: %d, resetting", err);
+	}
+
+	LOG_DBG("Waiting for PA synced");
+
+	err = k_sem_take(&sem_pa_synced, SEM_TIMEOUT);
+	if (err != 0) {
+		LOG_DBG("sem_pa_synced timed out, resetting");
+	}
+
+	LOG_DBG("Broadcast source PA synced, creating Broadcast Sink");
+
+	err = bt_bap_broadcast_sink_create(pa_sync, broadcaster_broadcast_id, &broadcast_sink);
+	if (err != 0) {
+		LOG_DBG("Failed to create broadcast sink: %d", err);
+	}
+
+	LOG_DBG("Broadcast Sink created, waiting for BASE");
+
+	err = k_sem_take(&sem_base_received, SEM_TIMEOUT);
+	if (err != 0) {
+		LOG_DBG("sem_base_received timed out, resetting");
+	}
+
+	LOG_DBG("BASE received, waiting for syncable");
+
+	err = k_sem_take(&sem_syncable, SEM_TIMEOUT);
+	if (err != 0) {
+		LOG_DBG("sem_syncable timed out, resetting");
+	}
+
+	/* sem_broadcast_code_received is also given if the
+	 * broadcast is not encrypted
+	 */
+	LOG_DBG("Waiting for broadcast code");
+
+	err = k_sem_take(&sem_broadcast_code_received, SEM_TIMEOUT);
+	if (err != 0) {
+		LOG_DBG("sem_broadcast_code_received timed out, resetting");
+	}
+
+	values_compare();
+
+	LOG_DBG("Waiting for BIS sync request");
+
+	err = k_sem_take(&sem_bis_sync_requested, SEM_TIMEOUT);
+	if (err != 0) {
+		LOG_DBG("sem_bis_sync_requested timed out, resetting");
+	}
+
+	sync_bitfield = bis_index_bitfield & requested_bis_sync;
+	stream_count = 0;
+	for (int i = 1; i < BT_ISO_MAX_GROUP_ISO_COUNT; i++) {
+		if ((sync_bitfield & BIT(i)) != 0) {
+			stream_count++;
+		}
+	}
+
+	LOG_DBG("Syncing to broadcast with bitfield: 0x%08x = 0x%08x "
+		"(bis_index) & "
+		"0x%08x "
+		"(req_bis_sync), stream_count = %u",
+		sync_bitfield, bis_index_bitfield, requested_bis_sync, stream_count);
+
+	err = bt_bap_broadcast_sink_sync(broadcast_sink, sync_bitfield, streams_p,
+					 sink_broadcast_code);
+	if (err != 0) {
+		LOG_DBG("Unable to sync to broadcast source: %d", err);
+	}
+
+	LOG_DBG("Waiting for stream(s) started");
+
+	err = k_sem_take(&sem_big_synced, SEM_TIMEOUT);
+	if (err != 0) {
+		LOG_DBG("sem_big_synced timed out, resetting");
+	}
+
+	LOG_DBG("Waiting for PA disconnected");
+	k_sem_take(&sem_pa_sync_lost, K_FOREVER);
+
+	LOG_DBG("Waiting for sink to stop");
+
+	err = k_sem_take(&sem_broadcast_sink_stopped, SEM_TIMEOUT);
+	if (err != 0) {
+		LOG_DBG("sem_broadcast_sink_stopped timed out, resetting");
+	}
+}
+
+int nac_test_main(void)
+{
+	int err;
+
+	err = init();
+	if (err) {
+		LOG_DBG("Init failed (err %d)", err);
+		return 0;
```

Comment:
Why does this return 0 on failure?

Reply:
Good question, it did in the sample but that was when this was main(). Changed to `return err;`